src/Symfony/Component/HttpClient/ConditionalHttpClient.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+use Symfony\Contracts\HttpClient\ResponseStreamInterface;
+
+/**
+ * @author Anthony Martin <anthony.martin@sensiolabs.com>
+ *
+ * @experimental in 4.3
+ */
+class ConditionalHttpClient implements HttpClientInterface
+{
+    private $options;
+
+    private $client;
+
+    /**
+     * @param HttpClientInterface $client
+     * @param array               $options should contain an array with regexp as key to filter the url and an array of options as value to be given for the HttpClient request() function
```

Comment:
e.g.
`@param array[] $options The default options to use when the regexp provided as key matches the requested URL.`